version: '2.1'
services:
  back:
    build: back-end/.
    image: back-python
    container_name: back-conabio
    ports:
      - 3000:3000
    volumes:
      - ./back-end/app:/back-end
    depends_on:
      db:
        condition: service_healthy
    command: ['python3', 'main.py']
  front:
    build: front-end/.
    image: front-nodejs
    container_name: front-conabio
    ports:
      - '3001:3000'
    volumes:
      - ./front-end/app/src:/front-end/src
    depends_on:
      db:
        condition: service_healthy
    command: ['npm', 'start']
  db:
    image: mdillon/postgis
    container_name: db-conabio
    hostname: db-hostname
    ports:
      - '5434:5432'
    volumes:
      - ./shapefiles:/shapefiles
      - ./sh-files:/docker-entrypoint-initdb.d
    environment:
      POSTGRES_PASSWORD: 123
      POSTGRES_USER: postgres
      POSTGRES_DB: conabio
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U postgres']
      interval: 10s
      timeout: 5s
      retries: 5
